aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -98,124 +108,275 @@ module AptosFramework::Stake {
-        let current_time = Timestamp::now_seconds();
-        assert!(current_time + MINIMUM_LOCK_PERIOD < locked_until_secs, Errors::invalid_argument(ELOCK_TIME_TOO_SHORT));
-        let addr = Signer::address_of(account);
-        let stake_pool = borrow_global_mut<StakePool>(to);
-        assert!(!find_delegation_from_pool(stake_pool, addr), Errors::invalid_argument(EDELEGATION_ALREADY_EXIST));
-        let delegation = Delegation {
-            coins,
-            rewards: Coin::zero<TestCoin>(),
-            locked_until_secs,
-        };
-        // add to pending_active if it's a current validator otherwise add to active directly
-        stake_pool.next_epoch_voting_power = stake_pool.next_epoch_voting_power + Coin::value<TestCoin>(&delegation.coins);
-        let maximum_stake = borrow_global<ValidatorSet>(@CoreResources).maximum_stake;
-        assert!(stake_pool.next_epoch_voting_power <= maximum_stake, Errors::invalid_argument(EDELEGATION_EXCEED_MAX));
-        if (!is_current_validator(to)) {
-            IterableTable::add(&mut stake_pool.active, addr, delegation);
-        } else {
-            IterableTable::add(&mut stake_pool.pending_active, addr, delegation);
-        }
+    /// TestCoin capabilities, set during genesis and stored in @CoreResource account.
+    /// This allows the Stake module to mint rewards to stakers.
+    struct TestCoinCapabilities has key {
+        mint_cap: MintCapability<TestCoin>,
     }
 
-    /// Unlock from active delegation, it's moved to pending_inactive if locked_until_secs < current_time or
-    /// directly inactive if it's not from an active validator.
-    public(script) fun unlock(account: &signer, from: address) acquires StakePool, ValidatorSet {
-        let addr = Signer::address_of(account);
-        let current_time = Timestamp::now_seconds();
-        let stake_pool = borrow_global_mut<StakePool>(from);
-        let delegation = withdraw_internal(&mut stake_pool.active, addr);
-        let is_current_validator = is_current_validator(from);
-        stake_pool.next_epoch_voting_power = stake_pool.next_epoch_voting_power - Coin::value<TestCoin>(&delegation.coins);
-        if (!is_current_validator) {
-            // move to inactive directly if it's not from an active validator
-            IterableTable::add(&mut stake_pool.inactive, addr, delegation);
-        } else if (delegation.locked_until_secs < current_time) {
-            // move to pending_inactive if it can be unlocked
-            IterableTable::add(&mut stake_pool.pending_inactive, addr,  delegation);
-        } else {
-            abort Errors::invalid_argument(EWITHDRAW_NOT_ALLOWED)
-        };
+    /// Initialize validator set to the core resource account.
+    public(friend) fun initialize_validator_set(account: &signer, minimum_stake: u64, maximum_stake: u64) {
+        SystemAddresses::assert_core_resource(account);
+        move_to(account, ValidatorSet {
```

Comment:
Similar to below, this likely is being installed on the core resource, so who cares if someone installs this on their own account?

Reply:
That's true. There's also that SystemAddresses::assert_core_resource(account); check as well that I'm leaving here for now. Eventually that will get removed when we remove the root account.